include/arc_utilities/filesystem.hpp

```diff
@@ -0,0 +1,35 @@
+#ifndef FILESYSTEM_HPP
+#define FILESYSTEM_HPP
+
+#include <iostream>
+#include <boost/filesystem.hpp>
+#include "arc_utilities/arc_exceptions.hpp"
+
+namespace arc_utilities
+{
+    void CreateDirectory(const boost::filesystem::path& p)
+    {
+        if (!boost::filesystem::is_directory(p))
+        {
+            std::cout << "\x1b[33;1m" << p << " does not exist! Creating ... ";
+
+            // NOTE: create_directories should be able to return true in this case
+            // however due to a bug related to a trailing '/' this is not currently
+            // the case in my version of boost
+            // https://svn.boost.org/trac/boost/ticket/7258
+            boost::filesystem::create_directories(p);
+            if (boost::filesystem::is_directory(p))
+    //            if (boost::filesystem::create_directories(p))
+            {
+                std::cout << "Succeeded!\x1b[37m\n";
```

Comment:
I don't like my utilities to print to std::cout.
Especially when they print even if everything goes as smoothly as possible.

Reply:
Fair enough, maybe add a verbose flag that defaults to false?

Reply:
verbose flag sounds good to me.

(p.s. emoji in github reviews feels weird)